Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -248,11 +234,14 @@ void OnValueChanged(string key, object value)
 			OnValuesChanged(new KeyValuePair<string, object>(key, value));
 		}
 
-		void OnValuesChanged(params KeyValuePair<string, object>[] values)
+		void OnValuesChanged(KeyValuePair<string, object> pair)
 		{
-			if (values == null || values.Length == 0)
-				return;
-			ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(values));
+			ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(new[] { pair }));
+		}
+
+		void OnValuesChanged(IEnumerable<KeyValuePair<string, object>> pairs)
```

Comment:
this, and the removal to `.ToArray()` invocation might be worth it, but I'm quite it was done for a reason. I have to evaluate if the reason still stands or if we can accept this part of the patch

Reply:
Seems, that ToArray call was really required by signature of method
```javascript
        void OnValuesChanged(params KeyValuePair<string, object>[] values)
         {
            if (values == null || values.Length == 0)
                return;
            ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(values));
        }
```
which may be safely spitted on two
```javascript
       void OnValuesChanged(KeyValuePair<string, object> pair)
       {
            ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(new[] { pair }));
       }

       void OnValuesChanged(IEnumerable<KeyValuePair<string, object>> values)
       {
            if (values == null || !values.Any())
                return;
            ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(values.Select(v => v));
        }
```
We may use projection 'values.Select(v => v)' to avoid sending of reference to original Merged Resource Dictionary into event handler.
Also we can remove checks because it is not necessary. Even if we add empty Resource Dictionary, nothing fear do not happens.

In result we have beauty code
```javascript
       void OnValuesChanged(KeyValuePair<string, object> pair) =>
            ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(new[] { pair }));

       void OnValuesChanged(IEnumerable<KeyValuePair<string, object>> values) =>
            ValuesChanged?.Invoke(this, new ResourcesChangedEventArgs(values.Select(v => v));
```